collaboration/sig-core/proposals/test-execution-using-labels.md

```diff
@@ -0,0 +1,93 @@
+* take backup
+* spinup new cluster and restore
+* install `Testdefinition` with parameter indicating after restore test
+* create `ClusterTestSuite` referencing the found `TestDefinitions` and wait for success
+
+
+For upgrade tests the situation is as follows:
+
+* install old kyma
+* install upgrade test `helm-chart`
+  * this creates a pod that does all the preparation work (compared to the backup test, this one does not use a pre-upgrade `TestDefinition`)
+* upgrade kyma
+* find **all** `TestDefinitions` in cluster
+* create `ClusterTestSuite` referencing the found `TestDefinitions` and wait for success
+
+Backup/Restore and upgrade tests implement their own framework for executing multiple tests.Event though both tests should be very similar and support reusing test cases it is very hard to do so. This leads to tests that are implemented multiple times.
+
+## Goal
+
+* It should be easy to reuse tests for multiple scenarios.
+* It should be easy to implement tests without having to use yet another test runner. Octopus should be the only external requirement.
+* It should be possible execute an unlimited number of test-cases per scenario
+* There should be one approach to run those tests from within the shell scripts. This reduces code in the shell scripts to scenario setup code.
+
+## Suggested solution
+
+Octopus allows to specify label queries for `ClusterTestSuites`. A defined set of labels should indicate the which `TestDefinition` to execute for which scenario.
+
+###The labels:
+
+each `TestDefinition` should be labeled with one or multiple of these labels:
+
+* `kyma-project.io/test.before-backup=true`
+* `kyma-project.io/test.after-restore=true`
+* `kyma-project.io/test.before-upgrade=true`
+* `kyma-project.io/test.after-upgrade=true`
+* `kyma-project.io/test.integration=true`
+
+>*This list can be extended at will.*
+
+Shell scripts install `ClusterTestSuites` with corresponding label queries e.g.:
+
+```yaml
+apiVersion: testing.kyma-project.io/v1alpha1
+kind: ClusterTestSuite
+metadata:
+  name: before-backup
+spec:
+  concurrency: 1
+  count: 1
+  selectors:
+    matchLabels:
+    - kyma-project.io/test.before-backup: true
+```
+
+
+## Implementation
+
+>*Octopus` does not require any changes*
+
+### Required Changes
+
+* Kyma CLI
+  * list available scenarios
+  * support creation of `ClusterTestSuite` based on scenario(s)
+* Existing Tests:
+  * apply labels to all `TestDefinitions` including the ones that are not part of a basic Kyma installation (e.g. tests-folder)
+* CI-Scripts
+  * remove custom test execution logic and replace by kyma-cli call
+* Upgrade and Backup/Restore tests:
+  * get rid off old upgrade test framework / backup/restore test framework by refactoring the implemented tests into their own test definitions
+
+
+### Optional Changes
+
+* move additional `Helm-charts` from tests-folder into default installation (resources-folder)
+  >*introduce additional/optional component 'additional-tests'*
```

Comment:
```suggestion
  * Introduce an additional/optional component named `additional-tests`.
```
Is this 
1. another optional change, or 
2. a way you can move those Helm charts you talk about in the line above? 

If `1.`, please merge that with the sentence above and get rid of the italics. 
If `2.`, please make it another bullet point and get rid of the italics (see the suggestion).

Reply:
its one way to add all those charts to the resources folder. 
I dont know if there is a better approach to do so so

Reply:
I see. In that case, I'd go with `1.`. I've made a suggestion.